scripts/provisioning/setupcmd.py

```diff
@@ -51,25 +57,48 @@ def s3confstore(self) -> str:
 
   def read_ldap_credentials(self):
     """Get 'ldapadmin' user name and password from confstore."""
-    prov_config_yaml = None
 
     try:
-      with open(self.s3_prov_config) as provconf:
-        prov_config_yaml = yaml.safe_load(provconf)
+      localconfstore = S3CortxConfStore(f'yaml://{self.s3_prov_config}', 'read_ldap_credentialsidx')
 
-      s3cipher_obj = CortxS3Cipher(None, False, 0, prov_config_yaml['CONFSTORE_OPENLDAP_CONST_KEY'])
+      s3cipher_obj = CortxS3Cipher(None, False, 0, localconfstore.get_config('CONFSTORE_OPENLDAP_CONST_KEY'))
       cipher_key = s3cipher_obj.generate_key()
 
-      encrypted_ldapadmin_pass = self.s3confstore.get_config(prov_config_yaml['CONFSTORE_LDAPADMIN_PASSWD_KEY'])
+      encrypted_ldapadmin_pass = self.s3confstore.get_config(localconfstore.get_config('CONFSTORE_LDAPADMIN_PASSWD_KEY'))
       self.ldap_passwd = s3cipher_obj.decrypt(cipher_key, encrypted_ldapadmin_pass)
 
-      self.ldap_user = self.s3confstore.get_config(prov_config_yaml['CONFSTORE_LDAPADMIN_USER_KEY'])
-    except IOError as ioe:
-      sys.stderr.write(f'failed to open config file: {self.s3_prov_config}, err: {ioe}\n')
-      raise ioe
-    except YAMLError as ye:
-      sys.stderr.write(f'yaml load failed for config file: {self.s3_prov_config}, err: {ye}\n')
-      raise ye
+      self.ldap_user = self.s3confstore.get_config(localconfstore.get_config('CONFSTORE_LDAPADMIN_USER_KEY'))
+
+      encrypted_rootdn_pass = self.s3confstore.get_config(localconfstore.get_config('CONFSTORE_ROOTDN_PASSWD_KEY'))
+      self.rootdn_passwd = s3cipher_obj.decrypt(cipher_key, encrypted_rootdn_pass)
+
     except Exception as e:
-      sys.stderr.write(f'unknown exception: {e}\n')
+      sys.stderr.write(f'exception: {e}\n')
       raise e
+
+  def read_cluster_id(self):
+    """Get 'cluster>cluster_id' from confstore."""
+
+    try:
+      localconfstore = S3CortxConfStore(f'yaml://{self.s3_prov_config}', 'read_cluster_ididx')
+      self.cluster_id = self.s3confstore.get_config(localconfstore.get_config('CONFSTORE_CLUSTER_ID_KEY'))
+    except Exception as e:
+      raise S3PROVError(f'exception: {e}\n')
+
+  def write_cluster_id(self, op_file: str = "/opt/seagate/cortx/s3/s3backgrounddelete/s3_cluster.yaml"):
+    """Set 'cluster>cluster_id' to op_file."""
+
+    try:
+      with open(f'{op_file}', 'w+') as fhndle:
+        fhndle.write("cluster_config:\n")
```

Comment:
this will override everything in the file, including the copyright header.  Try to see how to do "sed" in python.

Reply:
Addressed this using confstore to write the value. Please review the latest changes.

Reply:
FYI:
confstore save API is also removing the comment out copyright info. Created https://jts.seagate.com/browse/EOS-18356 to track this issue as discussed.